mods/default/craftitems.lua

```diff
@@ -88,6 +88,36 @@ minetest.register_craftitem("default:book_written", {
 	on_use = book_on_use,
 })
 
+minetest.register_craft({
+	type = "shapeless",
+	output = "default:book_written",
+	recipe = { "default:book", "default:book_written" }
+})
+
+minetest.register_on_craft(function(itemstack, player, old_craft_grid, craft_inv)
+	if itemstack:get_name() ~= "default:book_written" then
+		return
+	end
+
+	local copy = ItemStack("default:book_written")
+	local original
+	local index
+	for i = 1, 9 do
```

Comment:
I guess this should not be hardcoded. It's entirely possible to have bigger (and smaller) crafting grids (e.g. workbench mod).

I guess `player:get_inventory():get_size("craft")` should be better. (this hardcodes the "craft" list name, but I see no other way given the limited info you get in the `on_craft` callback, and shouldn't be a problem either)